composer.json

```diff
@@ -10,19 +11,25 @@
             "email": "mark.sagikazar@gmail.com"
         }
     ],
+    "provide": {
+        "php-http/async-client-implementation": "*",
+        "php-http/client-implementation": "*",
+        "psr/http-client-implementation": "*",
+        "psr/http-factory-implementation": "*",
+        "psr/http-message-implementation": "*"
```

Comment:
if the plugin is disabled or discovery only required indirectly, this would be incorrect, right? leading to confusing errors at runtime when discovery is used.

but i guess this can't be handled by the plugin, so we have to do it like this, right?

Reply:
Correct if the plugin is disabled, but then ppl decided to do that. Existing apps won't be affected since they already have an implementation.
For indirect dep, the plugin will install the missing implem if any, so the declaration is still right.